integration_tests/test_binary_compatibility.py

```
@@ -0,0 +1,604 @@
+        tuple: (cronos instance, binaries_path)
+    """
+    path = tmp_path_factory.mktemp("binary-compat")
+    port = 26300
+    configdir = Path(__file__).parent
+
+    # Build the Nix package containing both binaries
+    nix_file = configdir / f"configs/{nix_package}.nix"
+    print(f"Building Nix package: {nix_file}")
+    cmd = ["nix-build", nix_file]
+    print(*cmd)
+    subprocess.run(cmd, check=True)
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)